docarray/array/mixins/find.py

```diff
@@ -140,6 +139,12 @@ def find(
                 return result[0]
             else:
                 return result
+        elif isinstance(query, (DocumentArray, Document)):
+
+            if isinstance(query, Document):
+                query = DocumentArray(query)
+
+            _query = query.embeddings
         else:
             _query = query
```

Comment:
1. After the long if-else, `_query` could be `Dict` (when query is Dict and filter is Dict as well) or `ArrayType`. Previous code will only be `ArrayType`. This could be a potensial bug, because it assert that the type of `_query` will only be `ArrayType` in `_find` function. 
2. I don't think this long if-else logic readable, at least to me. It's clear before because we only need to detect the type of one parameter but now two. Can we refactor somehow? For example, use nested if-else.